enterprise_subsidy/apps/content_metadata/api.py

```diff
@@ -124,21 +124,52 @@ def get_geag_variant_id_for_content(self, content_identifier, content_data):
                 variant_id = additional_metadata.get('variant_id')
         return variant_id
 
+    def enroll_by_date_for_content(self, content_data, content_mode):
+        """
+        Determines the enrollment deadline for the given ``content_data``,
+        which could be either a course or course run, hence the branching
+        on content type below.
+
+        Args:
+            content_data: A dictionary of metadata pertaining to a course or course run.
+            content_mode: The (already computed) course mode for the content.
+
+        Returns:
+            A datetime string, or possibly null.
+        """
+        if content_data.get('content_type') == 'course':
+            return course_metadata.get('normalized_metadata', {}).get('enroll_by_date')
+
+        if content_mode == CourseModes.EDX_VERIFIED.value:
+            seats = content_data.get('seats', [])
+            seats_for_mode = [
+                seat for seat in content_data.get('seats', [])
+                if seat.get('type') == content_mode
+            ]
+            if seats_for_mode:
+                seat = seats_for_mode[0]
+                return seat.get('upgrade_deadline')
+            return None
+        elif content_mode == CourseModes.EXECUTIVE_EDUCATION.value:
+            return content_data.get('enrollment_end')
```

Comment:
For EE course run data, we can find the enrollment deadline in `enrollment_end`, rather than using the legacy approach of finding the `additional_metadata -> registration_deadline` at the _course_ level.
https://2u-internal.atlassian.net/wiki/spaces/CAT/pages/389873733/Consolidating+Internal+External+Courses+Life+Cycle#Move-relevant-course-run%2Fvariant-dates-from-Additional-Metadata-to-CourseRun